languages/eo.lua

```diff
@@ -1,4 +1,81 @@
 }
+
+SILE.doTexlike([[%
+\define[command=tableofcontents:title:eo]{Enhavoj}%
+\define[command=book:chapter:pre:eo]{Ĉapitro №}%
+]])
+
+local function digits(n)
+  local i, ret = -1
+  return function()
+    i, ret = i + 1, n % 10
+    if n > 0 then
+      n = math.floor(n / 10)
+      return i, ret
+    end
+  end
+end
+
+-- Esperanto name of a number
+local eo_string = function (num)
+  local words = {"unu", "du", "tri", "kvar", "kvin", "ses", "sep", "ok", "naŭ",  [0] =""}
+  local levels = {"mil", "miliono", "miliardo", [0]=""}
+  local ten = "dek"
+  local hundred = "cent"
+  for i=2,8,1 do
+    table.insert(levels, 2+i, words[i].."iliono")
+    table.insert(levels, 2+i+1, words[i].."iliardo")
+  end
+  -- main loop
+  local level = false
+  local function getname(pos, dig) --stateful, but effective.
+    level = level or (pos % 3 == 0)
+    if(dig == 0) then return "" end
+    local dig_name
+    if(dig == 1 and pos ~= 0) then
+      dig_name = ""
+    else
+      dig_name = words[dig]
+    end
+    local name = (pos % 3 == 2 and dig_name .. " " .. hundred .. " " or dig_name .. " ") .. (pos % 3 == 1 and ten .. " " or "")
+    if(level) then name, level = name .. levels[math.floor(pos / 3)] .. ((dig > 1 and pos > 5) and "j" or "") .. (pos > 5 and " kaj" or "") .. " ", false end
+    return name
+  end
+
+  local vword = ""
+
+  for i, v in digits(num) do
+    vword = getname(i, v) .. vword
+  end
+
+  vword = vword:gsub(" kaj $", " ")
+  while vword[#vword] == " " do
+    vword = vword:sub(1, #vword-1)
+  end
+  while vword[1] == " " do
+    vword = vword:sub(2, #vword)
+  end
+  return num == 0 and "nul" or vword
+end
+
+SU.formatNumber.eo = {
+  string = function (num)
+    return eo_string(num)
+  end,
+  nth = function (num)
+    return num .. 'a'
+  end
+}
+--[[
+SU.debug("eo", eo_string(3333300030))
```

Comment:
Can we add a `busted` style unit test for these? If there isn't one already I'll work one up for Turkish so there is an easy copy/edit model to follow.

Reply:
I don't think there is one :) the English versions aren't tested either fyi

Reply:
I see that! I just added one to this PR using the strings from the body of this PR. It looks like there is some extra whitespace being added, perhaps the `eo_string()` function could be refined to not pad everything, or (a bit lazier) at least trim it all off after assembly.

Reply:
> It looks like there is some extra whitespace being added

Really? It does trim though